pkg/kvdpa/device.go

```diff
@@ -197,38 +198,32 @@ GetVdpaDevicesByMgmtDev returns the VdpaDevice objects whose MgmtDev
 has the given bus and device names.
 */
 func GetVdpaDevicesByMgmtDev(busName, devName string) ([]VdpaDevice, error) {
-	result := []VdpaDevice{}
-	devices, err := ListVdpaDevices()
-	if err != nil {
-		return nil, err
-	}
-	for _, device := range devices {
-		if device.MgmtDev() != nil &&
-			device.MgmtDev().BusName() == busName &&
-			device.MgmtDev().DevName() == devName {
-			result = append(result, device)
-		}
-	}
-	if len(result) == 0 {
-		return nil, syscall.ENODEV
-	}
-	return result, nil
+	return listVdpaDevicesWithBusDevName(busName, devName)
 }
 
 /*ListVdpaDevices returns a list of all available vdpa devices */
 func ListVdpaDevices() ([]VdpaDevice, error) {
+	return listVdpaDevices()
+}
+
+func listVdpaDevicesWithBusDevName(busName, devName string) ([]VdpaDevice, error) {
 	msgs, err := GetNetlinkOps().RunVdpaNetlinkCmd(VdpaCmdDevGet, syscall.NLM_F_DUMP, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	vdpaDevs, err := parseDevLinkVdpaDevList(msgs)
+	vdpaDevs, err := parseDevLinkVdpaDevList(busName, devName, msgs)
 	if err != nil {
 		return nil, err
 	}
 	return vdpaDevs, nil
 }
 
+func listVdpaDevices() ([]VdpaDevice, error) {
```

Comment:
Why do we need this extra single-line private function? It seems only used on one place....